Move "Android native tests" from Circle CI to Buildkite

mokagio opened the pull request:
Move running the Android tests for the `react-native-editor` and `react-native-bridge` in Gutenberg submodule from Circle CI to Buildkite.

Things to note:

- `react-native-editor` doesn't seem to have any tests in the source, as such, I'm guessing CI only works as a safety check that the package build?
- There is another package that matches the `react-native-` pattern in the submodule, `react-native-aztec`. That one, too, has no tests
- The tests are running on the `android-staging` queue, https://github.com/Automattic/buildkite-ci/pull/281. See also the discussion here on things to keep in mind if we consider upgrading the main `android` queue to that configuration https://github.com/Automattic/buildkite-ci/pull/265#issuecomment-1749585960.

At this time, I am unclear on the value of these CI tasks, see also p9ugOq-47C-p2#comment-7715 . I think it's worth migrating them for consistency, but I would urge diving deep into establishing:

- Whether there's value in running these checks at all
- Whether this repo's CI is the appropriate one or they would be better running in the Gutenberg one

---

PR submission checklist:

- [x] I have considered adding unit tests where possible. – N.A.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary. – N.A.

mokagio committed:
Move `react-native-editor` Android tests from Circle CI to Buildkite

mokagio committed:
Move `react-native-bridge` Android tests from Circle CI to Buildkite

mokagio commented:
@oguzkocer @jkmassel pinged you for a review and particularly interested in your opinion on whether it's worth rolling out the Android infra upgrade for these tests given the tradeoffs highlighted in https://github.com/Automattic/buildkite-ci/pull/265#issuecomment-1749585960

Something else I'm looking at which might tip the scale is running the Sauce Labs / Appium tests on the updated `android-staging`.

oguzkocer commented:
If completing that update is going to be useful, I think we should do it, because we are going to need to do it at some point anyway. I'll leave the final decision to @jkmassel, but @mokagio please let me know if you'd like my help with it.

---

I skimmed through the changes and it looks good to me. However, I don't want to properly review & approve at the moment because we shouldn't merge `android-staging` to `trunk`.

mokagio commented:
Just a note. Since Oguz's comment, https://github.com/wordpress-mobile/gutenberg-mobile/pull/6320#issuecomment-1779786641, all the infra issues (that we know of) have been addressed. It is now safe to merge this PR and the `android-staging` queue is officially running.

In the long run, we'll want to move this to a `android` queue or the upcoming [cluster](https://buildkite.com/docs/clusters/overview). But moving out of CircleCI is the priority right now.

geriux commented on .buildkite/pipeline.yml:
```diff
@@ -94,6 +94,28 @@ steps:
 
+  - label: React Native Editor Android Tests
+    command: |
+      npm ci --prefer-offline --no-audit --no-progress
```
I think for these tests we can do:

```
npm ci --prefix gutenberg --prefer-offline --no-audit --no-progress --ignore-scripts
```

Since we only need Gutenberg's dependencies, this will avoid checking for i18n things, Jetpack dependencies and more and run faster. This is not a blocker but thought it'd be nice to try to reduce the time it takes to install all of the dependencies.

geriux commented on .buildkite/pipeline.yml:
```diff
@@ -94,6 +94,28 @@ steps:
+
+  - label: React Native Bridge Android Tests
+    command: |
+      npm ci --prefer-offline --no-audit --no-progress
```
Similar to https://github.com/wordpress-mobile/gutenberg-mobile/pull/6320/files#r1428541521

geriux reviewed (approved):
LGTM! Thanks for working on this! I left some feedback but it's not a blocker to merge this PR. 🚀

oguzkocer commented on .buildkite/pipeline.yml:
```diff
@@ -94,6 +94,28 @@ steps:
+      - *git-cache-plugin
+      - *nvm_plugin
+    agents:
+      queue: android-staging
```
> Just a note. Since Oguz's comment, https://github.com/wordpress-mobile/gutenberg-mobile/pull/6320#issuecomment-1779786641, all the infra issues (that we know of) have been addressed. It is now safe to merge this PR and the android-staging queue is officially running.

> In the long run, we'll want to move this to a android queue or the upcoming [cluster](https://buildkite.com/docs/clusters/overview). But moving out of CircleCI is the priority right now.

I'll re-iterate [my comment here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/6320#issuecomment-1779786641), this time by requesting changes, so this PR is not merged until it's addressed.

We should not use `android-staging` in production, because that queue is meant for testing only. There is no guarantee that someone in the team doesn't run a different test using that queue and break the CI for any project that uses it in production.

If we really need to use a temporary queue in production, we should deploy it as a new queue to avoid it getting overwritten.

Hope that makes sense!

mokagio committed:
Use `android` queue instead of `android-staging` for unit tests

Now that everything has been validated as working correctly on staging
and that the production queue has been upgraded, we can move to it.

mokagio committed:
Add `--prefix gutenberg` and `--ignore-scripts` to the tests `npm ci`

See https://github.com/wordpress-mobile/gutenberg-mobile/pull/6320#discussion_r1428541521

Co-authored-by: Gerardo <gerardo.pacheco@automattic.com>

mokagio commented:
Closed in favor of #6521
